### ISSUE TITLE
Install Error

### ISSUE THREAD

**Issue by SeAL80s** [hlissner/doom-emacs#494]
### Observed behavior
just git clone ,then cp init.example.el init.el ,then make install,then the error message came up

Package refresh done
Installing ox-reveal
  FATAL ERROR: (error) Need package ‘org-20150330’, but only 9.1.9 is available
Installing centered-window-mode
  FATAL ERROR: (error) Package ‘centered-window-mode-’ is unavailable
Installing ob-redis
  FATAL ERROR: (error) Package ‘ob-redis-’ is unavailable
Installing auth-password-store
  FATAL ERROR: (error) Package ‘auth-password-store-’ is unavailable
Installing help-fns+
  FATAL ERROR: (error) Package ‘help-fns+-’ is unavailable
Finished!

### Steps to reproduce

1. Select these example steps,
2. Delete them,
3. And replace them with precise steps to reproduce your issue.

### System information

<details>

<summary>
Arch Linux x86_64
emacs 25.3.1

</summary>

```
Replace this line with the output of *one* of these commands:

+ `M-x doom/info` (from inside Emacs)
+ `DEBUG=1 make doctor` (command line)
```
DOOM Doctor
Running Emacs v25.3.1, commit 5dacbb7cb1c6ac246a9ccd15e6c4290def67757c

shell: /usr/bin/zsh
Compiled with:
  XPM JPEG TIFF GIF PNG RSVG IMAGEMAGICK SOUND GPM DBUS GCONF GSETTINGS
  NOTIFY ACL GNUTLS LIBXML2 FREETYPE M17N_FLT LIBOTF XFT ZLIB
  TOOLKIT_SCROLL_BARS GTK3 X11 MODULES
uname -a:
  Linux SeAL80s 4.14.30-1-MANJARO #1 SMP PREEMPT Sun Mar 25 19:47:20 UTC
  2018 x86_64 GNU/Linux

Attempt to load DOOM: success! Loaded v2.0.9
Revision 5dacbb7cb1c6ac246a9ccd15e6c4290def67757c

----

test-emacs
test-windows
test-fonts
Warning: couldn’t find material-design-icons.ttf font in /home/daniel/.local/share/fonts/
  You can install it by running ‘M-x all-the-icons-install-fonts’ within
  Emacs.

  This could also mean you’ve installed them in non-standard locations,
  in which case, ignore this warning.
Warning: couldn’t find weathericons.ttf font in /home/daniel/.local/share/fonts/
  You can install it by running ‘M-x all-the-icons-install-fonts’ within
  Emacs.

  This could also mean you’ve installed them in non-standard locations,
  in which case, ignore this warning.
Warning: couldn’t find octicons.ttf font in /home/daniel/.local/share/fonts/
  You can install it by running ‘M-x all-the-icons-install-fonts’ within
  Emacs.

  This could also mean you’ve installed them in non-standard locations,
  in which case, ignore this warning.
Warning: couldn’t find fontawesome.ttf font in /home/daniel/.local/share/fonts/
  You can install it by running ‘M-x all-the-icons-install-fonts’ within
  Emacs.

  This could also mean you’ve installed them in non-standard locations,
  in which case, ignore this warning.
Warning: couldn’t find file-icons.ttf font in /home/daniel/.local/share/fonts/
  You can install it by running ‘M-x all-the-icons-install-fonts’ within
  Emacs.

  This could also mean you’ve installed them in non-standard locations,
  in which case, ignore this warning.
Warning: couldn’t find all-the-icons.ttf font in /home/daniel/.local/share/fonts/
  You can install it by running ‘M-x all-the-icons-install-fonts’ within
  Emacs.

  This could also mean you’ve installed them in non-standard locations,
  in which case, ignore this warning.
test-gnutls
test-tls
Validated https://elpa.gnu.org
Validated https://melpa.org
Successfully rejected https://self-signed.badssl.com
Successfully rejected https://wrong.host.badssl.com/
test-tar
Everything seems fine, happy Emacs’ing!

</details>

**Comment by hlissner** [comment]
Please switch to `develop`. These issues have been fixed and are due to be merged into master soon.

Also, please don't close this thread. I'd like to keep it open until `develop` is merged.

**Comment by liuxuzzz** [comment]
Me too..
When I checkout develop..
Recursive load: "/home/liuxu/.emacs.d/modules/config/private/init.el", "/home/liuxu/.config/doom/init.el", "/home/liuxu/.emacs.d/modules/config/private/init.el", "/home/liuxu/.config/doom/init.el", "/home/liuxu/.emacs.d/modules/config/private/init.el", "/home/liuxu/.config/doom/init.el", "/home/liuxu/.emacs.d/modules/config/private/init.el", "/home/liuxu/.config/doom/init.el", "/home/liuxu/.emacs.d/modules/config/private/init.el", "/home/liuxu/.emacs.d/init.el"
Makefile:92: recipe for target '.local/autoloads.el' failed
make: *** [.local/autoloads.el] Error 255

**Comment by hlissner** [comment]
Are you on the latest develop? The `config/private` module has been removed and is no longer needed. Delete your `~/.emacs.d/init.el` before updating to the latest Doom as it has been replaced with a default one.

**Comment by liuxuzzz** [comment]
liuxu:~$ git clone -b develop https://github.com/hlissner/doom-emacs ~/.emacs.dCloning into '/home/liuxu/.emacs.d'...
remote: Counting objects: 33294, done.
remote: Compressing objects: 100% (143/143), done.
remote: Total 33294 (delta 136), reused 180 (delta 106), pack-reused 33044
Receiving objects: 100% (33294/33294), 9.37 MiB | 2.07 MiB/s, done.
Resolving deltas: 100% (23004/23004), done.
Checking connectivity... done.
liuxu:~$ cd .emacs.d/
liuxu:.emacs.d$ git branch -av
* develop                    a582bdf app/calendar: fix typo causing void-variable error
  remotes/origin/HEAD        -> origin/master
  remotes/origin/develop     a582bdf app/calendar: fix typo causing void-variable error
  remotes/origin/master      5dacbb7 Merge branch 'develop'
  remotes/origin/screenshots e2956e5 Add screenshots
liuxu:.emacs.d$ cp init.example.el init.el
liuxu:.emacs.d$ make install
Symbol’s function definition is void: doom!
Makefile:93: recipe for target '.local/autoloads.el' failed
make: *** [.local/autoloads.el] Error 255

**Comment by liuxuzzz** [comment]
```bash
liuxu:~$ git clone -b develop https://github.com/hlissner/doom-emacs ~/.emacs.dCloning into '/home/liuxu/.emacs.d'...
remote: Counting objects: 33294, done.
remote: Compressing objects: 100% (143/143), done.
remote: Total 33294 (delta 136), reused 180 (delta 106), pack-reused 33044
Receiving objects: 100% (33294/33294), 9.37 MiB | 2.07 MiB/s, done.
Resolving deltas: 100% (23004/23004), done.
Checking connectivity... done.
liuxu:~$ cd .emacs.d/
liuxu:.emacs.d$ git branch -av
* develop                    a582bdf app/calendar: fix typo causing void-variable error
  remotes/origin/HEAD        -> origin/master
  remotes/origin/develop     a582bdf app/calendar: fix typo causing void-variable error
  remotes/origin/master      5dacbb7 Merge branch 'develop'
  remotes/origin/screenshots e2956e5 Add screenshots
liuxu:.emacs.d$ cp init.example.el init.el
liuxu:.emacs.d$ make install
Symbol’s function definition is void: doom!
Makefile:93: recipe for target '.local/autoloads.el' failed
make: *** [.local/autoloads.el] Error 255
```

**Comment by hlissner** [comment]
> cp init.example.el init.el

Change this step to `cp init.example.el ~/.doom.d/init.el`

Doom now supplies a default init.el that should not be overwritten. The documentation will be updated to reflect this soon.

**Comment by liuxuzzz** [comment]
```bash
liuxu:~$ git clone -b develop https://github.com/hlissner/doom-emacs ~/.emacs.d
Cloning into '/home/liuxu/.emacs.d'...
remote: Counting objects: 33294, done.
remote: Compressing objects: 100% (143/143), done.
remote: Total 33294 (delta 136), reused 180 (delta 106), pack-reused 33044
Receiving objects: 100% (33294/33294), 9.37 MiB | 2.05 MiB/s, done.
Resolving deltas: 100% (23004/23004), done.
Checking connectivity... done.
liuxu:~$ mkdir .doom.d
liuxu:~$ cd .emacs.d/
liuxu:.emacs.d$ cp init.example.el ~/.doom.d/init.el
liuxu:.emacs.d$ make install
Installing core packages
Importing package-keyring.gpg...
Importing package-keyring.gpg...done
Contacting host: elpa.gnu.org:443
Contacting host: elpa.gnu.org:443
Contacting host: melpa.org:443
Contacting host: melpa.org:443
Contacting host: orgmode.org:443
Contacting host: orgmode.org:443
Package refresh done
✓ Installed persistent-soft
✓ Installed use-package
✓ Installed quelpa
✓ Installed async
Installing core packages...done
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Recursive ‘require’ for feature ‘core’
Makefile:93: recipe for target '.local/autoloads.el' failed
make: *** [.local/autoloads.el] Error 255
```

**Comment by hlissner** [comment]
Odd, this only occurs if you have this line in `~/.doom.d/init.el`:

```elisp
(require 'core (concat user-emacs-directory "core/core"))
```

But it was removed from `init.example.el` in 0026f08e. Check to make sure that line is not present in `~/.doom.d/init.el`, and delete it if it is.

**Comment by liuxuzzz** [comment]
```bash
liuxu:.doom.d$ cat init.el | grep "concat"
liuxu:.doom.d$ 
```

**Comment by hlissner** [comment]
Is `~/.doom.d` empty besides the init.el?

Have you byte-compiled `~/.emacs.d` (with `make compile`)?

**Comment by liuxuzzz** [comment]
```bash
liuxu:~$ rm -rf .emacs.d/ .doom.d/
liuxu:~$ mkdir .doom.d
liuxu:~$ git clone -b develop https://github.com/hlissner/doom-emacs ~/.emacs.d
Cloning into '/home/liuxu/.emacs.d'...
remote: Counting objects: 33309, done.
remote: Compressing objects: 100% (153/153), done.
remote: Total 33309 (delta 145), reused 190 (delta 111), pack-reused 33044
Receiving objects: 100% (33309/33309), 9.38 MiB | 1.97 MiB/s, done.
Resolving deltas: 100% (23013/23013), done.
Checking connectivity... done.
liuxu:~$ cd .emacs.d/
liuxu:.emacs.d$ cp init.example.el ~/.doom.d/init.el
liuxu:.emacs.d$ git branch -av
* develop                    0e12761 Update README with new make quickstart workflow
  remotes/origin/HEAD        -> origin/master
  remotes/origin/develop     0e12761 Update README with new make quickstart workflow
  remotes/origin/master      5dacbb7 Merge branch 'develop'
  remotes/origin/screenshots e2956e5 Add screenshots
liuxu:.emacs.d$ ls -al ~/.doom.d/
total 16
drwxrwxr-x  2 liuxu liuxu 4096 4月   4 18:30 .
drwxr-xr-x 46 liuxu liuxu 4096 4月   4 18:30 ..
-rw-rw-r--  1 liuxu liuxu 6521 4月   4 18:30 init.el
liuxu:.emacs.d$ make install
Installing core packages
Importing package-keyring.gpg...
Importing package-keyring.gpg...done
Contacting host: elpa.gnu.org:443
Contacting host: elpa.gnu.org:443
Contacting host: melpa.org:443
Contacting host: melpa.org:443
Contacting host: orgmode.org:443
Package refresh done
✓ Installed persistent-soft
✓ Installed use-package
✓ Installed quelpa
✓ Installed async
Installing core packages...done
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")
Recursive ‘require’ for feature ‘core’
Makefile:97: recipe for target '.local/autoloads.el' failed
make: *** [.local/autoloads.el] Error 255
liuxu:.emacs.d$
`

**Comment by hlissner** [comment]
I cannot reproduce this. Do you have a `~/.config/doom` folder, by chance? In which you might possibly have an init.el that calls `(require 'core (concat user-emacs-directory "core/core"))`?

`~/.config/doom` is used instead of `~/.doom.d` if it is detected.

**Comment by liuxuzzz** [comment]
Haha,I install success.I am Ubuntu16.04,all configs in ```~/.config/```.Thank you ~

**Comment by SeAL80s** [comment]
now in emacs 25.3.3 with Arch Linux x86_64 . in doom-emacs master still has the problem :  
FATAL ERROR: (error) Need package ‘org-20150330’, but only 9.1.9 is available
FATAL ERROR: (error) Package ‘centered-window-mode-’ is unavailable
FATAL ERROR: (error) Package ‘ob-redis-’ is unavailable

and in develop branch 
![screenshot_2018-04-13_11-37-31](https://user-images.githubusercontent.com/12682385/38715678-225cd37a-3f0f-11e8-91b2-fa7f53740d48.png)
do not have the cool UI ,nothing .but it says "Doom loaded 123 packages across 0 modules in 0.585s"

**Comment by mbark** [comment]
@SeAL80s did you run `make` before starting emacs on the develop branch?

**Comment by SeAL80s** [comment]
@mbark In the Master branch 'make install ' will produce 
> "FATAL ERROR: (error) Need package ‘org-20150330’, but only 9.1.9 is available"

git checkout develop then make (or make install ) It appears like above , no color UI and It's just 44 packages installed . In the Master Branch It's 300+ .

And ,If 'make install '  in master branch ,then checkout develop , 'make' will remove 250+ packages ,remain 44 packages.

If git clone doom-emacs , then cp init.example.el init.el (in develop branch ,there is a init.el already) , 'make install ' will produce Error 'Symbol’s function definition is void: doom!'

**Comment by fuxialexander** [comment]
@SeAL80s The correct flow is 
1. clone doom-emacs
2. `git checkout develop`
3. `mkdir ~/.doom.d && cp ~/.emacs.d/init.example.el ~/.doom.d/init.el`
4. `cd ~/.emacs.d && make install`

**Comment by mbark** [comment]
@SeAL80s @hlissner has mentioned in some other issues that the docs will be updated and that it is part of the next release.

**Comment by SeAL80s** [comment]
@hlissner In the develop branch there is a init.el in .emacs.d calls (require 'core (concat user-emacs-directory "core/core")) which cause @liuxuzzz problem . My solution is in the master branch ,
> mkdir ~/.doom.d ,cp init.example.el ~/.doom.d/init.el 
then 
> git checkout develop 
switch to develop branch 
> mv ~/.doom.d/init.el ~/.emacs.d/init.el 
to overwrite the default init.el in develop branch ,which contain only one line 
> (require 'core (concat user-emacs-directory "core/core")) .
Then make install works perfectly .

**Comment by hlissner** [comment]
@SeAL80s The order is incorrect. Please do not use `init.example.el` from the master branch if you intend to use `develop`. This will cause problems. I highly recommend you do _not_ modify or replace `~/.emacs.d/init.el` unless you understand the repercussions of doing so.

+ ~/.emacs.d/init.el should _only_ contain `(require 'core (concat user-emacs-directory "core/core"))`
+ ~/.doom.d/init.el should **not** contain `(require 'core (concat user-emacs-directory "core/core"))`

This is all you need to install Doom:

### On `master`

```sh
git clone https://github.com/hlissner/doom-emacs ~/.emacs.d
cd ~/.emacs.d
cp init.example.el init.el
make install
```

### On `develop` (recommended)

```sh
git clone https://github.com/hlissner/doom-emacs ~/.emacs.d
cd ~/.emacs.d
mkdir ~/.doom.d
cp init.example.el ~/.doom.d/init.el
make install
```

**Comment by SeAL80s** [comment]
@hlissner But Your order doesn't for me . On master there is this error 
> FATAL ERROR: (error) Need package ‘org-20150330’, but only 9.1.9 is available

and On develop cp init.example.el init.el , then 'make install' will show ":doom is invalid " .
The order i mentioned works for me . I don't know why , but 'make install' will sucess .
By the way ,on develop there is already has a init.el ,if cp init.example.el ~/.doom.d/init.el will  show
> Warning: Eager macro-expansion skipped due to cycle:
  … => (load "core.el") => (macroexpand-all …) => (macroexpand (eval-and-compile …)) => (load "init.el") => (load "core.el")
Eager macro-expansion failure: (error "Recursive ‘require’ for feature ‘core’")

**Comment by hlissner** [comment]
> On develop cp init.example.el init.el, then...

`cp init.example.el init.el` is incorrect. 

`cp ~/.emacs.d/init.example.el ~/.doom.d/init.el` is correct.

> On develop there is already has a init.el

This is correct. It should not be modified. `~/.emacs.d/init.el` should only contain:

```el
(require 'core (concat user-emacs-directory "core/core"))
```

The above line *must not be inside* `~/.doom.d/init.el`! Or you get that error.

As I said before, do `cp ~/.emacs.d/init.example.el ~/.doom.d/init.el`.

**Comment by jameskolce** [comment]
Hi @hlissner 
I just made a fresh install of Emacs 26.1 but was getting errors when installing Doom, so I found this issue and followed the next steps:

```
git clone https://github.com/hlissner/doom-emacs ~/.emacs.d
cd ~/.emacs.d
git checkout develop
mkdir ~/.doom.d
cp init.example.el ~/.doom.d/init.el
make install
```

but I still get:

```
.emacs.d ) make install

Symbol's function definition is void: advice-add
make: *** [.local/autoloads.el] Error 255
```

Any advice?
Thanks.

**Comment by hlissner** [comment]
@jameskolce Make sure that the emacs you have in your `PATH` is the correct one. If `advice-add` is undefined, that implies you are running a version of an earlier version of Emacs (24 or under).

Run `emacs --version` in your shell to find out.

**Comment by jameskolce** [comment]
@hlissner Thanks! I found the problem: `emacs --version` was returning the right version (26.1) so I didn't know what could be the problem, but at the end it was indeed trying to use Emacs 22 as you mention, because I had made a command alias instead of a link. It's working now, thanks.

**Comment by BenJWard** [comment]
I'm having issues on develop in particular with emacs-snippets, and php-extras: 

```
N82106:.emacs.d bward$ make install
Looking for packages to install...
2 packages will be installed:

+ emacs-snippets (QUELPA)
+ php-extras (QUELPA)

Proceed? (y or n) y
Installing php-extras
✕ ERROR ((wrong-type-argument arrayp nil))
Installing emacs-snippets
✕ ERROR ((wrong-type-argument arrayp nil))
Finished!
```

I used process 
```
git clone https://github.com/hlissner/doom-emacs ~/.emacs.d
cd ~/.emacs.d
git checkout develop
mkdir ~/.doom.d
cp init.example.el ~/.doom.d/init.el
make install
```

**Comment by hlissner** [comment]
@BenJWard I've been hunting for this bug for a while now, but still haven't come up with any leads. Would you be able to recreate the error with `DEBUG=1 make install`, then copy the resulting backtrace (as much of it as you can) into a pastebin to send to me?

**Comment by BenJWard** [comment]
@hlissner I've just tried that and seems to hang in places wherever it says "entering debugger..."

```
Extracting expand-region-20171217.2125/ruby-mode-expansions.el
Extracting expand-region-20171217.2125/expand-region-custom.el
Extracting expand-region-20171217.2125/latex-mode-expansions.el
Extracting expand-region-20171217.2125/jsp-expansions.el
Extracting expand-region-20171217.2125/js-mode-expansions.el
Extracting expand-region-20171217.2125/expand-region-core.el
Extracting expand-region-20171217.2125/text-mode-expansions.el
Extracting expand-region-20171217.2125/python-el-expansions.el
Extracting expand-region-20171217.2125/the-org-mode-expansions.el
Entering debugger...
```

**Comment by tidux** [comment]
I see that exact same hang on a fresh install from develop today.

**Comment by hlissner** [comment]
I'll close this, since the original issue has been resolved.

@tidux @BenJWard Could you create a new issue for the hang? Please include your `M-x doom/info`.
